### PR TITLE
Adds ability to copy user email addresses from staff list

### PR DESCRIPTION
**Pull request by johnclary**
## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/13507

## Testing
**URL to test:**  [Netlify](https://deploy-preview-1110--atd-moped-main.netlify.app/)

1. Navigate to the Staff page
2. Find your user account and edit it so that you are a MUG member and also add a note. Save.
3. Find you user account and click the edit icon to verify your changes were saved. Now return to the staff list.
4. Add a filter to staff list: **MUG Member** **equals**  **Yes**
6. Now use the menu above the top-left of the staff list to copy all Moped User Group email addresses.
7. Paste your clipboard somewhere and observe that the email addresses you pasted match the filtered staff list.
8. Now use the same menu to copy **All active users**. Paste your clipboard and observer it is a long list that only includes active users who do not have the **non-login user** role.

---
#### Ship list
- [x] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/staff/EditStaffView.js [hunk 1]
```diff
@@ -67,6 +67,8 @@ const EditStaffView = () => {
       password,
       roles,
       workgroup_id,
+      is_user_group_member,
+      note,
```
Comment (johnclary) [inline moped-editor/src/views/staff/EditStaffView.js:71]: this fixes a bug i introduced [here](https://github.com/cityofaustin/atd-moped/pull/1096). it is not currently possible to edit these two user fields in Moped unless the user a non-login user :/

File: moped-editor/src/views/staff/StaffListViewCustomComponents.js [hunk 2]
```diff
@@ -30,3 +40,118 @@ export const EditUserButton = ({id}) => (
...
+      setCopiedListName(null);
+      handleCloseMenu();
+    }, 300);
+    return () => clearTimeout(timeout);
+  }, [copiedListName, handleCloseMenu]);
+
+  const menuOpen = Boolean(anchorEl);
+
+  return (
+    <div>
```
Comment (chiaberry) [inline moped-editor/src/views/staff/StaffListViewCustomComponents.js:114]: dunno if you want to give all users the ability to download the emails or only admins. i guess limiting who can see the button would just make the staff view "cleaner"

**Review by chiaberry — APPROVED** [review]
copied email addresses!

File: moped-editor/src/views/staff/StaffListViewCustomComponents.js [hunk 3]
```diff
@@ -30,3 +40,118 @@ export const EditUserButton = ({id}) => (
...
+  useEffect(() => {
+    /**
+     * Effect which closes the menu after a brief pause
+     */
+    if (!copiedListName) return;
+    const timeout = setTimeout(() => {
+      setCopiedListName(null);
+      handleCloseMenu();
+    }, 300);
+    return () => clearTimeout(timeout);
```
Comment (mddilley) [inline moped-editor/src/views/staff/StaffListViewCustomComponents.js:108]: love seeing a useEffect with cleanup - don't get to do this often 💯

**Review by mddilley — APPROVED** [review]
Tested this out on the preview, and I like the UI flow! All the expected emails copied to my clipboard ✅ 🚢

**Review by roseeichelmann — CHANGES_REQUESTED** [review]
When i click on the copy buttons nothing is copied to my clipboard and i get the error "ReferenceError: ClipboardItem is not defined". I am testing on Firefox :0

[simplescreenrecorder-2023-08-28_11.45.17.webm](https://github.com/cityofaustin/atd-moped/assets/67130963/993adde0-4e03-412f-8b31-52ee3c1af88e)

**Comment by chiaberry** [comment]
wont work on firefox :( https://developer.mozilla.org/en-US/docs/Web/API/ClipboardItem#browser_compatibility

**Comment by johnclary** [comment]
i have re-requested review from all after making a few tweaks to the button labels to make the behavior more clear. i think it's fine to leave this button visible to all users 😅

File: moped-editor/src/views/staff/StaffListViewCustomComponents.js [hunk 4]
```diff
@@ -49,12 +49,15 @@ export const EditUserButton = ({ id }) => (
...
+ * Filter function to identify all active users who are not non-login users and
+ * do not have a `+` in their email address
  * @param {object} user - a `moped_user` object
  * @returns user
  */
 const allUserFilter = (user) =>
-  !user.is_deleted && !user.roles.includes("non-login-user");
+  !user.is_deleted &&
+  !user.roles.includes("non-login-user") &&
+  user.email.indexOf("+") < 0;
```
Comment (johnclary) [inline moped-editor/src/views/staff/StaffListViewCustomComponents.js:60]: this excludes our test accounts e.g. "transportation.data+mopedadmin@austintexas.gov"

**Review by chiaberry — APPROVED** [review]
the menu and buttons are more clear this way, i like it

**Review by roseeichelmann — APPROVED** [review]
yay, works on firefox now! and i also like the updates to the buttons :ship:

**Comment by johnclary** [comment]
thank you!!!

**Review by mddilley — APPROVED** [review]
Didn't retest the UI thoroughly but tested and reviewed the new commits, and this looks great! 🙏

**Comment by johnclary** [comment]
thakns @mddilley for taking another look as well!

**Pull request merged**